map/gps_track_collection.cpp

```diff
@@ -63,6 +51,27 @@ std::pair<size_t, size_t> GpsTrackCollection::Add(std::vector<TItem> const & ite
     if (!m_items.empty() && m_items.back().m_timestamp > item.m_timestamp)
       continue;
 
+    if (m_items.empty())
+    {
+      m_trackInfo.m_maxElevation = item.m_altitude;
+      m_trackInfo.m_minElevation = item.m_altitude;
+    }
+    else
+    {
+      auto const & lastItem = m_items.back();
+      m_trackInfo.m_length += mercator::DistanceOnEarth(lastItem.GetPoint(), item.GetPoint());
+      m_trackInfo.m_duration = item.m_timestamp - m_items.front().m_timestamp;
```

Comment:
Always calculating this info even if it's not visible may consume more battery. Is there a way to do a lazy initialization or even better, to calc only when there's an UI subscriber?

Or is there an idea to always display this info on the screen while recording?

Reply:
This info will be **always** visible before track is saved. But the screens are not implemented yet.
There are no reason to delay the calculations because this data will be presented to the user is the "saving screen" (PP for the track recording process). All the iteration over the track points will be done in both cases. Its simpler and faster solution to keep the stats up to date - the location updates are not so fast (1-1.5 sec). The calculations. are not complex and stats struct is a lightweight.

This data will be used in 2 places:
1). PP for the track recording (not stated to implement it yet but the design is under development). 
The User may keep it open to see the current stats or may close the screen and leave only the activity indication (red button somewhere) that opens the PP.
<img width="280" alt="image" src="https://github.com/user-attachments/assets/b7be8789-1ce5-4efd-a82c-1d2a93ad8119" />

2) Live Activity widget (is under development) Ive tested it a little bit - its quite useful thing!
https://github.com/organicmaps/organicmaps/pull/9482

<img width="312" alt="image" src="https://github.com/user-attachments/assets/98471e69-3cdf-4932-a682-8247c77b5ac8" />

Reply:
Good points, thanks!